tests/test_browser.py

```diff
@@ -4917,14 +4917,20 @@ def test_wasm2js_fallback(self):
       self.run_browser('test.html', 'hello!', '/report_result?0')
 
   @no_fastcomp('wasm-backend specific feature')
-  def test_minimal_runtime_wasm2js_fallback(self):
-    src = 'src.cpp'
-    create_test_file(src, self.with_report_result(open(path_from_root('tests', 'small_hello_world.c')).read()))
-    self.compile_btest([src, '-s', 'WASM=2', '-s', 'MINIMAL_RUNTIME=1', '-o', 'test.html'])
+  def test_wasm2js_fallback_on_wasm_compilation_failure(self):
+    for args in [[], ['-s', 'MINIMAL_RUNTIME=1']]:
```

Comment:
I think most of us (in particular @sbc100) prefer this kind of thing be done using `@parameterized` instead of a loop. That's easier to debug (one function per mode, so if you know the failing function name, you know which mode failed, unlike the loop where it can be less easy to find which iteration it was).

Reply:
Yeah I was't going to say anything because I know @juj  tends to disagree, but I do find these tests frustrating because they can fail so many different ways, and the line number of the failures doesn't tell you which case failed...   haveing the "_minimal" version fail separately is so much nicer IMHO.

Since this pattern occurs elsewhere in this file I'd be OK with landing this as is and cleaning them both up as a followup..

Reply:
Sounds good. A drawback I find with parameterized is that it does not work quite as well from command line to launch the tests. With this mode I can do `python tests/runner.py browser.test_wasm2js_fallback_on_wasm_compilation_failure` and it will test both of the modes. For some reason `python tests/runner.py browser.test_wasm2js_fallback_on_wasm_compilation_failure*` does not work for parameterized tests.

Reply:
Are you sure? I constantly do things like `tests/runner.py other.test_metadce*` to run all the parameterized metadce tests, etc. Do you have an example of a parameterized test where things don't work atm with wildcards?

Reply:
Indeed I relay on the `test_foo*` syntax to run all the parameterized tests at once.

In parallel no less :)

Reply:
On my Mac I get e.g.

```
zclb:master clb$ python tests/runner.py other.test_check_sourcemapurl_default*
tests/runner.py:WARNING: use EMTEST_ALL_ENGINES=1 in the env to run against all JS engines, which is slower but provides more coverage
Test suites:
['test_other']
Running test_other: (2 tests)
Using 2 parallel test processes
test_check_sourcemapurl_default_profiling (test_other.other) ... ERROR
Process Process-1:
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/clb/emsdk/emscripten/master/tests/parallel_testsuite.py", line 33, in g_testing_thread
    result.addError(test, e)
  File "/Users/clb/emsdk/emscripten/master/tests/parallel_testsuite.py", line 155, in addError
    self.buffered_result = BufferedTestError(test, err)
  File "/Users/clb/emsdk/emscripten/master/tests/parallel_testsuite.py", line 163, in __init__
    exctype, value, tb = err
ValueError: need more than 1 value to unpack
test_check_sourcemapurl_default_normal (test_other.other) ... ERROR
Process Process-2:
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/clb/emsdk/emscripten/master/tests/parallel_testsuite.py", line 33, in g_testing_thread
    result.addError(test, e)
  File "/Users/clb/emsdk/emscripten/master/tests/parallel_testsuite.py", line 155, in addError
    self.buffered_result = BufferedTestError(test, err)
  File "/Users/clb/emsdk/emscripten/master/tests/parallel_testsuite.py", line 163, in __init__
    exctype, value, tb = err
ValueError: need more than 1 value to unpack

DONE: combining results on main thread


----------------------------------------------------------------------
Ran 0 tests in 0.116s

OK
zclb:master clb$ 
```

Windows also fails, but with different error message iirc. (not at a Windows box right now to test)

Reply:
That looks like a python2/3 issue - it works on 3, but yes, I see it fails on 2.

This seems unrelated to parametrized - debugging it, the internal error in the parallel test runner is `'module' object has no attribute 'perf_counter'`. It happens when running the test runner on any two or more tests.

Fix in https://github.com/emscripten-core/emscripten/pull/11421